app/src/main/java/com/example/tasky/agenda/data/remote/util/Util.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.tasky.agenda.data.remote.util
+
+import android.annotation.SuppressLint
+import com.example.tasky.agenda.data.remote.model.AgendaResponse
+import com.example.tasky.agenda.presentation.model.Agenda
+import com.example.tasky.core.data.Resource
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+fun Resource<AgendaResponse>.mapToAgenda(): Resource<List<Agenda>> {
+    if (this is Resource.Error) {
+        return Resource.Error(this.errorMessage)
+    } else {
+        var agendaList: List<Agenda> = mutableListOf()
+
+        with(this as Resource.Success) {
+            agendaList = agendaList + this.data.events.map {
+                Agenda.Event(
+                    title = it.title,
+                    description = it.description,
+                    date = it.from.toCurrentTime(),
+                )
+            }
+
+            agendaList = agendaList + this.data.reminders.map {
+                Agenda.Reminder(
+                    title = it.title,
+                    description = it.description,
+                    date = it.remindAt.toCurrentTime(),
+                )
+            }
+
+            agendaList = agendaList + this.data.tasks.map {
+                Agenda.Task(
+                    title = it.title,
+                    description = it.description,
+                    date = it.remindAt.toCurrentTime(),
+                    isDone = it.isDone
+                )
+            }
+        }
+        return Resource.Success(agendaList)
+    }
+}
+
+@SuppressLint("NewApi")
+fun Long.toCurrentTime(): String {
+    return LocalDateTime.ofInstant(
+        Instant.ofEpochMilli(this),
+        ZoneId.systemDefault()
+    ).toLocalTime().toString()
+}
```

Comment:
You could have a "DateExt" class where you store all the extensions for date (Or a DateUtil class)

Having a big util class with different type of stuff (Date and Agenda) makes it hard to find where the methods are

Reply:
And in which package it belongs if I only have extension functions? This one uses Android/Java stuff so does ut belong to domain or data?

Reply:
You are only using them in data
So they can be kept in data